packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts

```diff
@@ -119,13 +138,38 @@ export class SetNewPoolOracleReaderSubAction extends SubAction<SetNewPoolOracleR
           functionName: 'maxStaleTimes',
           args: [oracle.token],
         })
-        if (maxStaleTime !== 0n && priceId !== zeroAddress) {
+        if (maxStaleTime !== 0n || priceId !== zeroHash) {
           throw new ContractValidateError('PythOracleReader Token parameters has been already set')
         }
         break
       }
       case 'lsdApi3': {
-        //TODO: handle oracle has already been set
+        const params = oracle.params as LsdApi3Params
+        // check lsd api3
+        const lsdApi3ProxyOracleReaderArtifact = await readArtifact('LsdApi3ProxyOracleReader')
+        const [dataFeedProxy, quoteToken, maxStaleTime] = await this.client.publicClient.readContract({
+          address: oracle.oracleReader,
+          abi: lsdApi3ProxyOracleReaderArtifact.abi,
+          functionName: 'dataFeedInfos',
+          args: [oracle.token],
+        })
+        if (maxStaleTime !== 0n || dataFeedProxy !== zeroAddress || quoteToken !== zeroAddress) {
+          throw new ContractValidateError('LsdApi3ProxyOracleReader Token parameters has been already set')
+        }
+        // check if not params for the quote token, quote token oracle shoud be already set
+        if (!params.quoteToken.params) {
+          if (!registry.api3ProxyOracleReaderProxy) throw new ValidateInputValueError('registry: api3ProxyOracleReader not found')
+          const api3ProxyOracleReaderArtifact = await readArtifact('Api3ProxyOracleReader')
+          const [dataFeedProxy, maxStaleTime] = await this.client.publicClient.readContract({
+            address: registry.api3ProxyOracleReaderProxy,
+            abi: api3ProxyOracleReaderArtifact.abi,
+            functionName: 'dataFeedInfos',
+            args: [params.quoteToken.token],
+          })
+          // check if token parameters has been set
+          if (maxStaleTime === 0n && dataFeedProxy === zeroAddress)
```

Comment:
should we check `if (maxStaleTime === 0n || dataFeedProxy === zeroAddress)`

Reply:
the comment was wrong